### DISCUSSION TITLE
Using a PicoW to Stream Music

### DISCUSSION THREAD

**Discussion by Adventurer21** [micropython/micropython#14151] [RP2040 / Pico]
Hi Guys,

I have a project to stream music using a PicoW as the audio driver.  I'm concerned that doing this with Micropython might be a bit much for the Pico.  I would probably use use one core for the audio output which leaves the other core to handle the wifi traffic, decoding of the mp3 packets, and user control inputs.

Is this feasible?  What do you think?

**Comment by peterhinch** [comment]
The only solution I know for decoding MP3 is with external hardware, the VS1053. See [this repo](https://github.com/peterhinch/micropython-vs1053).

**Comment by gneverov** [comment]
[CircuitPython](https://docs.circuitpython.org/en/latest/shared-bindings/audiomp3/index.html) has a Python wrapper of a C MP3 decoder library.

**Reply by bixb922** [reply]
Also see discussion #11307
